project/src/js/ui/tournament/detail/tournamentDetailCtrl.js

```diff
@@ -1,56 +1,83 @@
 import socket from '../../../socket';
+import socketHandler from './socketHandler';
 import * as utils from '../../../utils';
 import * as xhr from '../tournamentXhr';
 import helper from '../../helper';
 import m from 'mithril';
 
 export default function controller() {
-  const id = m.route.param('id');
+	helper.analyticsTrackView('Tournament details');
 
-  helper.analyticsTrackView('Tournament details');
+	const tournament = m.prop();
+	const hasJoined = m.prop(false);
 
-  const tournament = m.prop();
+	function reload(data) {
+		tournament(data);
+		hasJoined(data.me && !data.me.withdraw);
 
-  function reload(data) {
-    tournament(data);
-    if (data.socketVersion) {
-      socket.setVersion(data.socketVersion);
-    }
-    m.redraw();
-  }
+		if (data.socketVersion) {
+			socket.setVersion(data.socketVersion);
```

Comment:
File looked fine for me locally, but I reindented anyway with atom. Hopefully that fixes the issue.